### PR TITLE
docs: describe steps for Private Voting use case.

### PR DESCRIPTION
**Pull request by 0xorbus**
Describe Private Voting steps and the roles in each step.

### CODE REVIEW

File: versioned_docs/version-V2/guides/identies.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+publicly, and are used by the zero-knowledge proof to check the identity commitment exists in the identity commitment tree and the nullifier map is checked
+so that the nullifiers hash corresponding to the identity nullifier was not used
+to broadcast a signal for the same external nullifier .
+An identity commitment on an identity id = (idpub, idnullifier, idtrapdoor) is computed as
+Commitment(id) = Commitidtrapdoor(idpub, idnullifier)
+where Commit is a Pedersen hash instantiated in 5.3.2
+
+(From whitepaper)
+
+- what's the "system" - the dApp?
```
Comment (cedoor) [inline versioned_docs/version-V2/guides/identies.md:17]: In that case the system was the smart contract I guess.

File: versioned_docs/version-V2/guides/index.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+A relay is a third-party who receives a fee for including relayed transactions in the blockchain. (McMenamin, Daza, and Fitz. https://eprint.iacr.org/2022/155.pdf, p.3)
+
+Ethereum transactions are tied to a specific address, and transactions sent from
+this address consume Ether to pay for gas. When a signal is broadcast, the
+Ethereum address used for the transaction might be used to break the anonymity
+of the broadcaster. Semaphore allows signals to be received from any address,
+allowing a relayer to broadcast a signal on behalf of a user. Applications might
+provide rewards for relayers and implement front-running prevention mechanisms, such as requiring the signals to include the relayer’s address, binding the
+signal to that specific address.(https://semaphore.appliedzkp.org/whitepaper-v1.pdf, p.6)
```
Comment (cedoor) [inline versioned_docs/version-V2/guides/index.md:11]: The paper refers to version 1, which in my opinion went beyond the protocol and tried to explain how the architecture of a dApp with Semaphore could be organized. The main problem is that the one who creates the zero-knowledge proof anonymously cannot be the same one who signs the Ethereum transaction, as the signer's address is public. The relayer may be an architectural solution.

File: versioned_docs/version-V2/use-cases/index.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+### Use Semaphore for private voting
+Walk me through https://raw.githubusercontent.com/appliedzkp/semaphore/main/circuits/scheme.png:
+
+Private voting use case:
+  - Goal is to clearly describe roles and responsibilities in each step.
+  - Comment on draft review https://github.com/semaphore-protocol/docs/pull/2/files#diff-ed5798bfec02cfeada5a9c8b8afd91462ad8b4a544793e16616342a79d51066e
+ 
+Identities:
+  - Is an identity commitment (Semaphore ID) created prior to voting or at the time of vote? In this scenario, is each vote a new commitment?
```
Comment (cedoor) [inline versioned_docs/version-V2/use-cases/index.md:59]: The identity commitment is created before. Steps:
1. Users generate their identity
2. Users join the group with their identity
3. Users who are part of the group generate their proofs and vote

Votes are signals (https://semaphore.appliedzkp.org/docs/technical-reference/intro).

File: versioned_docs/version-V2/use-cases/index.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+Groups:
+  - Arbitrary array of Semaphore IDs?
```
Comment (cedoor) [inline versioned_docs/version-V2/use-cases/index.md:61]: Groups are Merkle trees. Group members are identity commitments. Identity commitments are tree leaves.

File: versioned_docs/version-V2/use-cases/index.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  - Membership in a group?
+  - Who defines?
+  - When are they created?
+  - How are they created?
+
+Verifier:
+  - What is this doing? Gets all "verifiers" from the network?
```
Comment (cedoor) [inline versioned_docs/version-V2/use-cases/index.md:68]: No it's an `ethers` function to allow you to deploy contracts. A `Verifier` contract in this case.

File: versioned_docs/version-V2/use-cases/index.md (hunk 3 above)
Comment (0xorbus) [reply versioned_docs/version-V2/use-cases/index.md:59]: In the call, I want to talk through UI interactions for this in the most basic implementation--are 1 and 2 in a single interaction (e.g. "Register") or multiple?

File: versioned_docs/version-V2/use-cases/private-voting.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
+---
+sidebar_position: 2
+---
+
+## Private voting overview
+
```
Comment (0xorbus) [inline versioned_docs/version-V2/use-cases/private-voting.md:6]: @cedoor Still in draft, but I would appreciate your feedback on this `private-voting` page to make sure it's accurate.

File: versioned_docs/version-V2/use-cases/private-voting.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+Learn how Semaphore enables applications to do the following:
+
+- Register members as voters.
+- Allow members to vote anonymously.
+- Prove voter membership.
+- Record and prove votes.
```
Comment (cedoor) [inline versioned_docs/version-V2/use-cases/private-voting.md:12]: - Prove voter membership.
- Record and prove votes.

These two points could be included in the second one "- Allow members to vote anonymously."

File: versioned_docs/version-V2/use-cases/private-voting.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+- Prevent double-voting.
+
+## Steps
+
+Learn how your users and your application interact with Semaphore and Ethereum, in the following steps:
+
+1. Create a poll.
+2. Qualify members.
+2. Register ballot topics.
+3. Register voters.
```
Comment (cedoor) [inline versioned_docs/version-V2/use-cases/private-voting.md:22]: Shouldn't "Qualify members" and "Register voters" be the same step?

File: versioned_docs/version-V2/use-cases/private-voting.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+:::info
+Learn how to use [Ethers.js]() to an [check Ethereum wallets]().
+:::
+
+The Dapp allows the member to do the following:
+
+- [Create a poll](#create-a-poll).
+- [Register to vote](#register-voters).
+- [Vote on a topic](#record-votes).
```
Comment (cedoor) [inline versioned_docs/version-V2/use-cases/private-voting.md:48]: I'd write "Register voters" and "Record votes" here.

File: versioned_docs/version-V2/use-cases/private-voting.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+### Register voters
+
+A voter is a [member](#members) that joins a poll to vote.
+To join a poll, a member uses the Dapp for the following contract interactions:
+
+1. Member selects a poll (topic) to join.
+2. Dapp calls a contract function to create a private identity for the member.
+3. Wallet prompts the Member to sign the transaction and create the identity.
+4. Dapp passes the private identity to a contract function that creates a public anonymous Semaphore ID (identity commitment).
```
Comment (cedoor) [inline versioned_docs/version-V2/use-cases/private-voting.md:79]: In the following steps there is no interaction with contracts: 2, 3, 4. Users can generate their Semaphore identity off-chain, with the `@zk-kit/identity` package. They only need their Metamask wallet to sign a message (here a demo with the same feature with Interep: https://js.interep.link/identity/)

File: versioned_docs/version-V2/use-cases/private-voting.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+With the member registered for a poll, learn how the Dapp [records votes on a topic](#record-votes).
+
+### Record votes
+
+To vote on a topic, a member registers in a poll and votes on the poll topic.
+The member signs a vote with a private on-chain or off-chain identity that remains provable and anonymous.
+
+In the Dapp, when a member clicks to vote on a poll topic, the Dapp calls the contract to do the following:
+1. Create a proof that the member is registered for the poll and hasn't already voted in the poll. See how Semaphore [verifies proofs and prevents double-voting]().
```
Comment (cedoor) [inline versioned_docs/version-V2/use-cases/private-voting.md:91]: Step 1 happens off-chain, there is no interaction with contracts. `@zk-kit/protocols` is used to generate zk proofs off-chain.

File: versioned_docs/version-V2/use-cases/private-voting.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+2. Broadcast the member's vote. See how Semaphore [broadcasts signals anonymously](). 
+3. Prompt the member to sign the vote transaction.
+
+Once the member signs the transaction, the Dapp sends the vote proofs to a relayer that signs and relays the transactions to Ethereum.
```
Comment (cedoor) [inline versioned_docs/version-V2/use-cases/private-voting.md:95]: The member shouldn't sign the transaction, it must be signed by the relay.

File: versioned_docs/version-V2/guides/identities.md [hunk 13]
```diff
@@ -0,0 +1,17 @@
+3.2 Identities
+Users registering to the system possess an identity they must keep
+secret. A commitment to this structure is known to the public.
+An identity is a tuple (idpub, idnullifier, idtrapdoor)
+
+An identity is generated locally by a user registering to Semaphore, where idkey
+and idnullifier are sampled randomly. Both idpub and idnullifier are not revealed
+publicly, and are used by the zero-knowledge proof to check the identity commitment exists in the identity commitment tree and the nullifier map is checked
+so that the nullifiers hash corresponding to the identity nullifier was not used
```
Comment (rachelaux) [inline versioned_docs/version-V2/guides/identities.md:9]: test comment

**Comment by 0xorbus** [comment]
Closing and reopening with updated branch.

**Pull request closed without merging**